File: app/src/main/java/rqg/fantasy/open163/tv/business/main/MainActivityModule.java
package rqg.fantasy.open163.tv.business.main;

import dagger.Binds;
import dagger.Module;
import rqg.fantasy.open163.tv.ActivityScope;

/**
 * * Created by rqg on 04/07/2017.
 */

@Module
public abstract class MainActivityModule {

    @Binds
    @ActivityScope
    public abstract MainContract.View provideView(MainActivity activity);


    @Binds
    @ActivityScope
    public abstract MainContract.Presenter bindPresenter(MainPresenterImpl mainPresenter);
}

File: app/src/main/java/rqg/fantasy/open163/tv/business/main/MainActivityModule.kt
package rqg.fantasy.open163.tv.business.main

import dagger.Binds
import dagger.Module

/**
 * * Created by rqg on 04/07/2017.
 */


@Module
abstract class MainActivityModule {
    @Binds
    abstract fun bindView(mainActivity: MainActivity): MainContract.View

    @Binds
    abstract fun bindPresenter(mainPresenterImpl: MainPresenterImpl): MainContract.Presenter
}